Content.Server/GameObjects/Components/Interactable/MatchstickComponent.cs

```diff
@@ -0,0 +1,127 @@
+﻿#nullable enable
+using System.Threading.Tasks;
+using Content.Shared.Audio;
+using Content.Shared.GameObjects.Components;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Server.GameObjects;
+using Robust.Server.GameObjects.EntitySystems;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Components.Timers;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Interactable
+{
+    [RegisterComponent]
+    [ComponentReference(typeof(IHotItem))]
+    public class MatchstickComponent : Component, IHotItem, IInteractUsing, IAfterInteract
+    {
+        public override string Name => "Matchstick";
+
+        private MatchstickState _currentState;
+
+        /// <summary>
+        /// How long will matchstick last in seconds.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadOnly)] private int _duration;
+
+        /// <summary>
+        /// Sound played when you ignite the matchstick.
+        /// </summary>
+        private string? _igniteSound;
+
+        /// <summary>
+        /// Point light component.
+        /// </summary>
+        private PointLightComponent? _pointLightComponent;
+
+        /// <summary>
+        /// Current state to matchstick. Can be <code>Unlit</code>, <code>Lit</code> or <code>Burnt</code>.
+        /// </summary>
+        [ViewVariables]
+        public MatchstickState CurrentState
+        {
+            get => _currentState;
+            private set
+            {
+                _currentState = value;
+
+                if (_pointLightComponent != null)
+                {
+                    if (_currentState == MatchstickState.Lit)
+                    {
+                        _pointLightComponent.Enabled = true;
+                    }
+                    else
+                    {
+                        _pointLightComponent.Enabled = false;
+                    }
+                }
+
+                if (Owner.TryGetComponent(out AppearanceComponent? appearance))
+                {
+                    appearance.SetData(MatchstickVisual.Igniting, _currentState);
+                }
+            }
+        }
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            serializer.DataField(ref _duration, "duration", 10);
```

Comment:
Magic strings for serialization are fine in our codebase (although they will be mostly fixed when serializationv3 comes around, so that's something to look forwards to!)